### ISSUE TITLE
Unable to install QZCompanionNordictrackTreadmill on a Nordictrack C1650

### ISSUE THREAD

**Issue by daunce** [cagnulein/QZCompanionNordictrackTreadmill#106]
I'm unable to install QZCompanionNordictrackTreadmill on a Nordictrack C1650.

I can ping the treadmill OK.

After booting into safe mode, I use the web browser to download the file from a local web server (The browser doesn't support https),  but when running it from the downloaded location it displays an error. "`There is a problem parsing the package`".

![ProblemParsingPackage](https://github.com/user-attachments/assets/a8a42f5d-055c-4eb4-82a4-d870c645d2dd).

Even trying some of the steps from the guide:
Running QZ-Companion-simple.bat:
```
Ping statistics for 172.31.2.36:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 93ms, Maximum = 99ms, Average = 96ms

Waiting for 0 seconds, press a key to continue ...
disconnected everything
* daemon not running; starting now at tcp:5037
* daemon started successfully
cannot connect to 172.31.2.36:5555: No connection could be made because the target machine actively refused it. (10061)
List of devices attached


Waiting for 0 seconds, press a key to continue ...
- waiting for device -
```

This treadmill runs:
- Android 2.2
- iFit 2.19.5

I've tried various methods to get into privilege mode, but none are successful. I can get into safe mode to access the Android desktop by continously pressing the 4 horiontal lines button when the first Android logo is displayed after powering on the treadmill.

There is no pin hole to factory reset it. If the treadmill has wifi access, it can be factory reset within iFit maintenance settings.

Unknown sources is enabled. 
USB debugging is enabled.

![UnknownSources](https://github.com/user-attachments/assets/3c97fc28-e4f5-4a0a-97c2-b4b7672c6ddd)
![devOptions](https://github.com/user-attachments/assets/9dcdfe09-7411-4462-93c5-6e06224c998c)
![AboutDevice](https://github.com/user-attachments/assets/9bae1a12-cab8-40c3-b761-e766045176b6)

I'm hoping to use the treadmill with Zwift's structured training to automatically change the speed during interval sessions. 

Is there a minimum Android version that will run QZCompanionNordictrackTreadmill ?

Let me know if there's more information required.

It's amazing the work being put in to help out so many people. Great job.

**Comment by cagnulein** [comment]
hi, i guess for your treadmill you don't need to use the companion flow. you just need to set the proform ip setting in the nordictrack treadmill settings in qz 
Let me know!

**Comment by daunce** [comment]
Hi @cagnulein. I was just reviewing old closed issues and found a similar issue where you suggested the same thing.

I just got back from testing it. 

Setting the IP in the Proform IP section worked! I'm not sure if other settings are required, but it worked. 

![QZ iPhone app](https://github.com/user-attachments/assets/94ee0406-2239-4043-a74d-88dfbd775156)

I was able to see speed, incline, heart rate in the QZ app. Using the app I could also change the speed & incline of the treadmill.

Using Zwift on my iPad, I was able to connect to the QZ virtual device (my iPhone), and see all the same metrics in Zwift/QZ app. 

I tried a custom workout, but unfortunately Zwift wasn't able to control the speed of the treadmill to match the interval session. If I had read the Zwift FAQ on running I would have seen:

> Q: Will Zwift control my treadmill?
> A: No. Zwift doesn't control the incline or the speed of your treadmill. You can manually change your incline and your speed at your discretion.

And I just noticed there's already tiles for speed presets! That will do nicely!

I am very happy with the outcome though. Now I can log my runs. Thanks Roberto!
